homeassistant/components/syncthru/config_flow.py

```diff
@@ -0,0 +1,128 @@
+        return await self._async_check_and_create("user", user_input)
+
+    async def async_step_import(self, user_input=None):
+        """Handle import initiated flow."""
+        return await self.async_step_user(user_input=user_input)
+
+    async def async_step_ssdp(self, discovery_info):
+        """Handle SSDP initiated flow."""
+        await self.async_set_unique_id(discovery_info[ssdp.ATTR_UPNP_UDN])
+        self._abort_if_unique_id_configured()
+
+        url = discovery_info.get(
+            ssdp.ATTR_UPNP_PRESENTATION_URL,
+            f"http://{urlparse(discovery_info[ssdp.ATTR_SSDP_LOCATION]).hostname}/",
+        )
+        if any(x.data[CONF_URL] == url for x in self._async_current_entries()):
+            return self.async_abort(reason="already_configured")
+
+        name = discovery_info.get(ssdp.ATTR_UPNP_FRIENDLY_NAME)
+        if name:
+            # Remove trailing " (ip)" if present for consistency with user driven config
+            name = re.sub(r"\s+\([\d.]+\)\s*$", "", name)
+
+        # https://github.com/PyCQA/pylint/issues/3167
+        self.context.update(  # pylint: disable=no-member
+            {CONF_URL: url, CONF_NAME: name, "title_placeholders": {CONF_NAME: name}}
+        )
+        return await self.async_step_confirm()
+
+    async def async_step_confirm(self, user_input=None):
+        """Handle discovery confirmation by user."""
+        if user_input is not None:
+            return await self._async_check_and_create("confirm", user_input)
+
+        return await self._async_show_form(
+            step_id="confirm",
+            user_input={
+                # https://github.com/PyCQA/pylint/issues/3167
+                CONF_URL: self.context[CONF_URL],  # pylint: disable=no-member
+                CONF_NAME: self.context[CONF_NAME],  # pylint: disable=no-member
+            },
+        )
+
+    async def _async_show_form(self, step_id, user_input=None, errors=None):
+        """Show our form."""
+        if user_input is None:
+            user_input = {}
+        return self.async_show_form(
+            step_id=step_id,
+            data_schema=vol.Schema(
+                {
+                    vol.Required(CONF_URL, default=user_input.get(CONF_URL, "")): str,
+                    vol.Optional(CONF_NAME, default=user_input.get(CONF_NAME, "")): str,
+                }
+            ),
+            errors=errors or {},
+        )
+
+    async def _async_check_and_create(self, step_id, user_input):
+        """Validate input, proceed to create."""
+        user_input[CONF_URL] = url_normalize(
+            user_input[CONF_URL], default_scheme="http"
+        )
+        if "://" not in user_input[CONF_URL]:
+            return await self._async_show_form(
+                step_id=step_id, user_input=user_input, errors={CONF_URL: "invalid_url"}
+            )
+
+        await self.async_set_unique_id(user_input[CONF_URL])
+        self._abort_if_unique_id_configured()
```

Comment:
We should not abort if a user is initiating the flow. In that case finishing the flow will remove the other one, which is probably what they want.

Reply:
Done.